lib/Macros/Sources/ObservationMacros/Availability.swift

```diff
@@ -0,0 +1,148 @@
+  var availability: IfConfigClauseSyntax.Elements? {
+    switch self {
+    case .attributes(let attributes):
+      if let availability = attributes.availability {
+        return .attributes(availability)
+      } else {
+        return nil
+      }
+    default:
+      return nil
+    }
+  }
+}
+
+extension IfConfigClauseSyntax {
+  var availability: IfConfigClauseSyntax? {
+    if let availability = elements?.availability {
+      return IfConfigClauseSyntax(
+        leadingTrivia: leadingTrivia,
+        unexpectedBeforePoundKeyword,
+        poundKeyword: poundKeyword,
+        unexpectedBetweenPoundKeywordAndCondition,
+        condition: condition,
+        unexpectedBetweenConditionAndElements,
+        elements: availability,
+        unexpectedAfterElements,
+        trailingTrivia: trailingTrivia
+      )
+    } else {
+      return nil
+    }
+  }
+  
+  var clonedAsIf: IfConfigClauseSyntax {
+    IfConfigClauseSyntax(
+      leadingTrivia: leadingTrivia,
+      unexpectedBeforePoundKeyword,
+      poundKeyword: .poundIfKeyword(),
+      unexpectedBetweenPoundKeywordAndCondition,
+      condition: condition,
+      unexpectedBetweenConditionAndElements,
+      elements: elements,
+      unexpectedAfterElements,
+      trailingTrivia: trailingTrivia
+    )
+  }
+}
+
+extension IfConfigDeclSyntax {
+  var availability: IfConfigDeclSyntax? {
+    var elements = [IfConfigClauseListSyntax.Element]()
+    for clause in clauses {
+      if let availability = clause.availability {
+        if elements.isEmpty {
+          elements.append(availability.clonedAsIf)
+        } else {
+          elements.append(availability)
+        }
+      }
+    }
+    if elements.isEmpty {
+      return nil
+    } else {
+      return IfConfigDeclSyntax(
+        leadingTrivia: leadingTrivia,
+        unexpectedBeforeClauses,
+        clauses: IfConfigClauseListSyntax(elements),
+        unexpectedBetweenClausesAndPoundEndif,
+        poundEndif: poundEndif,
+        unexpectedAfterPoundEndif,
+        trailingTrivia: trailingTrivia
+      )
+    }
+    
+  }
+}
+
+extension AttributeListSyntax.Element {
+  var availability: AttributeListSyntax.Element? {
+    switch self {
+    case .attribute(let attribute):
+      if let availability = attribute.availability {
+        return .attribute(availability)
+      }
+    case .ifConfigDecl(let ifConfig):
+      if let availability = ifConfig.availability {
+        return .ifConfigDecl(availability)
+      }
+    default:
+      break
+    }
+    return nil
+  }
+}
+
+extension AttributeListSyntax {
+  var availability: AttributeListSyntax? {
+    var elements = [AttributeListSyntax.Element]()
+    for element in self {
+      if let availability = element.availability {
+        elements.append(availability)
+      }
+    }
+    if elements.isEmpty {
+      return nil
+    }
+    return AttributeListSyntax(elements)
+  }
+}
+
+extension DeclGroupSyntax {
+  var availability: AttributeListSyntax? {
```

Comment:
@DougGregor this seems really complex to forward things - should this be included in SwiftSyntax/macros directly?

Reply:
Perhaps. We'd need to clarify what exactly should be automatically transferred to the extensions and implement that in SwiftSyntax.